File: dotnet-mvc/CleanArchitecture/Framework/Web/Dto/ProductResponse.cs
namespace CleanArchitecture.Framework.Web.Dto;

/// <summary>
/// DTO for product responses. Belongs to the FRAMEWORK layer (outermost).
/// Maps from the domain entity to the API response format.
/// </summary>
public record ProductResponse(
    Guid Id,
    string Name,
    string Description,
    decimal Price,
    string Category,
    int StockQuantity,
    string Sku,
    bool Active,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

/// <summary>
/// Paginated response wrapper.
/// </summary>
public record PaginatedResponse<T>(
    IEnumerable<T> Content,
    int Page,
    int PageSize,
    int TotalElements,
    int TotalPages
);

File: java-mvc/clean-architecture/src/main/java/com/ecommerce/clean/framework/web/dto/ProductResponse.java
package com.ecommerce.clean.framework.web.dto;

import com.ecommerce.clean.entity.Product;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record ProductResponse(
        UUID id,
        String name,
        String description,
        BigDecimal price,
        String category,
        int stockQuantity,
        String sku,
        boolean active,
        boolean inStock,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
    public static ProductResponse from(Product product) {
        return new ProductResponse(
                product.getId(),
                product.getName(),
                product.getDescription(),
                product.getPrice(),
                product.getCategory(),
                product.getStockQuantity(),
                product.getSku(),
                product.isActive(),
                product.isInStock(),
                product.getCreatedAt(),
                product.getUpdatedAt()
        );
    }
}
